ietf/utils/xmldraft.py

```diff
@@ -60,17 +60,43 @@ def cleanup():  # called when context exited, even if there's an exception
                 tree.tree = v2v3.convert2to3()
         return tree, xml_version
 
-    def _document_name(self, anchor):
-        """Guess document name from reference anchor
-
-        Looks for series numbers and removes leading 0s from the number.
-        """
-        anchor = anchor.lower()  # always give back lowercase
-        label = anchor.rstrip('0123456789')  # remove trailing digits
-        if label in ['rfc', 'bcp', 'fyi', 'std']:
-            number = int(anchor[len(label):])
-            return f'{label}{number}'
-        return anchor
+    def _document_name(self, ref):
+        """Get document name from reference."""
+        series = ["rfc", "bcp", "fyi", "std"]
+        # handle xinclude first
+        # FIXME: this assumes the xinclude is a bibxml href
+        if ref.tag.endswith("}include"):
+            name = re.search(
+                rf"reference\.({'|'.join(series).upper()})\.(\d{{4}})\.xml",
+                ref.attrib["href"],
+            )
+            if name:
+                return f"{name.group(1)}{int(name.group(2))}".lower()
+            name = re.search(r"reference\.I-D\.(.*)\.xml", ref.attrib["href"])
```

Comment:
What if the reference was to a specific version already, e.g. `https://bib.ietf.org/public/rfc/bibxml3/reference.I-D.draft-ietf-sipcore-multiple-reasons-00.xml`?
Wouldn't this then return "draft-draft-ietf..."?

Reply:
Wouldn't the URL be `https://bib.ietf.org/public/rfc/bibxml3/reference.I-D.ietf-sipcore-multiple-reasons-00.xml`?

Reply:
no. In fact, until we broke what Tony had in mind, the URL you post would not have worked. His design (which was what was running at tools) used I-D.draft-<whatever> for versioned things and ID.<whatever> for unversioned references. We made it so that both forms work at the datatracker. There are lots of documents that use the older forms.

Reply:
Ah - fixed!